### PR TITLE
Simple warmup: default to opting out

### PR DESCRIPTION
**Pull request by oschaaf**
The current single request we issue for the simple warmup pollutes
some of the counters we emit in our output.

Looking forward, when landing generic phases, we might want to
default to opting out.
This new flag could then serve as an opt-in afterwards, for those
who want to preserve old behavior.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

File: api/client/options.proto [hunk 1]
```diff
@@ -187,4 +187,6 @@ message CommandLineOptions {
   repeated string labels = 28;
   // TransportSocket configuration to use in every request
   envoy.config.core.v3.TransportSocket transport_socket = 27;
+  // Do not perform the simple warmup call.
+  google.protobuf.BoolValue no_simple_warmup = 32;
```
Comment (mum4k) [inline api/client/options.proto]: We have learned internally that naming boolean flags in the negative can end up being confusing. E.g:

--no-simple-warmup=False

Can we name the flag simple-warmup instead and set its default accordingly?

Comment (mum4k) [inline api/client/options.proto]: On a similar topic - why wouldn't we make it an opt-in right away? What are the use cases to have this enabled by default?

File: source/client/client_worker_impl.h [hunk 2]
```diff
@@ -31,7 +31,7 @@ class ClientWorkerImpl : public WorkerImpl, virtual public ClientWorker {
                    const RequestSourceFactory& request_generator_factory,
                    Envoy::Stats::Store& store, const int worker_number,
                    const Envoy::MonotonicTime starting_time,
-                   Envoy::Tracing::HttpTracerPtr& http_tracer);
+                   Envoy::Tracing::HttpTracerPtr& http_tracer, const bool simple_warmup);
```
Comment (mum4k) [inline source/client/client_worker_impl.h]: How will this API evolve? This is about my usual concern around readability of boolean flags. Is this related to phases and will this be removed in favor of some phase information? Or should we introduce an enum now?

File: api/client/options.proto (hunk 1 above)
Comment (oschaaf) [reply api/client/options.proto]: fixed in 5940d5601eaa9924782c57be5a67895a7a1dfdd5

Comment (oschaaf) [reply api/client/options.proto]: I think that is a good idea, fixed in 5940d5601eaa9924782c57be5a67895a7a1dfdd5 (with a callout in the changelist)

File: source/client/client_worker_impl.h (hunk 2 above)
Comment (oschaaf) [reply source/client/client_worker_impl.h]: See 5940d5601eaa9924782c57be5a67895a7a1dfdd5, let me know what you think.

File: README.md [hunk 3]
```diff
@@ -77,6 +78,9 @@ format>
 
 Where:
 
+--simple-warmup
+Do not perform the simple warmup call.
```
Comment (mum4k) [inline README.md]: We should also update the comment next to the flag. Should now say something like:

Perform the simple warmup call.

While we are here, do we also want to expand this a bit to explain what simple warmup call is?

Comment (oschaaf) [reply README.md]: whoops, done in bb70db154224cb50d06d315eb7e3359b25a8be9f (thanks!)

**Pull request merged**